polars/polars-core/src/frame/row/transpose.rs

```diff
@@ -54,25 +61,56 @@ impl DataFrame {
                 }
                 let cols = buffers
                     .into_iter()
-                    .enumerate()
-                    .map(|(i, buf)| {
-                        let mut s = buf.into_series().cast(dtype).unwrap();
-                        s.rename(&format!("column_{i}"));
-                        s
-                    })
+                    .map(|buf| buf.into_series().cast(dtype).unwrap())
                     .collect::<Vec<_>>();
                 Ok(DataFrame::new_no_checks(cols))
             }
+        }?;
+        out.set_column_names(colnames)?;
+        match keep_names_as {
```

Comment:
Can we add this column before we transpose/ generate the columns.

Inserting at the start of a `Vec` is `O(n)` where it is free if we add it before and then extend that `vec` with the transposed columns.

Reply:
I think this might get a little awkward and we should only revisit if it seems like there are performance issues. I'm trying to create a vector of series upfront and then push to it, which seemed reasonable. But the `numeric_transpose` versions use parallel iterators, so I think I'd have to take the tail of this pre-allocated new Series vector, pass it mutably to `numeric_transpose` and do `collect_into_vec`. Not sure if there's a cleaner way to do this.

Reply:
The non-numeric one collects into a `vec`, so that iterator can extend to an existing buffer.

The numeric one, can reallocate by appending to the prepared `vec`. That has the same cost we have now.

Reply:
Ok, that's fine. Easy enough. It seems like this has been optimized pretty thoroughly since it's a slow operation, but that makes it tricky to change. Our of curiosity, do you know if there's a reasonable way to handle this with the parallel iterators? Could I use unsafe code to have the parallel iterator collect into a mutable alias of a prepared buffer?

Reply:
Apparently there's `par_extend` so I think there should be no reallocations anymore.

Reply:
> Apparently there's par_extend so I think there should be no reallocations anymore.

Even that one reallocates :) But that is definitely what we need in this case, as it is a relallocation less.